src/Diagnostics.RuntimeHost/Services/DiagnosticTranslator/DiagnosticTranslatorService.cs

```diff
@@ -0,0 +1,320 @@
+            {
+                RenderingType renderingType = diagnosticResponse.Dataset[j].RenderingProperties.Type;
+                // Starting from Insights and data summary localizations for now. Also render title and description for the remaining rendering types.
+                switch (renderingType)
+                {
+                    case RenderingType.Insights:
+                        diagnosticResponse.Dataset[j] = await GetInsightsTranslation(diagnosticResponse.Dataset[j], language);
+                        break;
+                    case RenderingType.DataSummary:
+                        diagnosticResponse.Dataset[j] = await GetDataSummaryTranslation(diagnosticResponse.Dataset[j], language);
+                        break;
+                    case RenderingType.DropDown:
+                        diagnosticResponse.Dataset[j] = await GetDropdownTranslation(diagnosticResponse.Dataset[j], language);
+                        break;
+                    default:
+                        diagnosticResponse.Dataset[j] = await GetBaseRenderingTranslation(diagnosticResponse.Dataset[j], language);
+                        break;
+                }
+            }
+
+            return diagnosticResponse;
+        }
+
+        public async Task<IEnumerable<DiagnosticApiResponse>> GetMetadataTranslations(IEnumerable<DiagnosticApiResponse> listDetectorsResponse, string language)
+        {
+            if (listDetectorsResponse == null || !IsLocalizationApplicable(language))
+            {
+                return listDetectorsResponse;
+            }
+
+            IEnumerable<DiagnosticApiResponse> translatedResponse = listDetectorsResponse;
+      
+            List<Definition> allDetectorsDefinitions = listDetectorsResponse.Select(detectorResponse => detectorResponse.Metadata).ToList<Definition>();
+            List<Definition> metadataTranslations = await GetDetectorsDefinitionsTranslations(allDetectorsDefinitions, language);
+
+            if (metadataTranslations != null && metadataTranslations.Count == translatedResponse.Count())
+            {
+                translatedResponse = translatedResponse.Select((response, i) => { response.Metadata = metadataTranslations[i]; return response; });
+            }
+
+            return translatedResponse;
+        }
+
+        #region Localization Helper Methods
+
+        // Scenarios to enable localization:
+        // 1. EnableLocalization is not set or set to be true in appsettings/keyvault
+        // 2. Language to translate to is not empty or English
+        private static bool IsLocalizationApplicable(string language)
+        {
+            return (string.IsNullOrWhiteSpace(_enableLocalization) || string.Compare(_enableLocalization, "true", StringComparison.OrdinalIgnoreCase) == 0) && !String.IsNullOrWhiteSpace(language) && string.Compare(language, "en", StringComparison.OrdinalIgnoreCase) != 0 && !language.StartsWith("en.", StringComparison.CurrentCulture);
```

Comment:
shouldnt this check be _enableLocalization != null && string.Compare(_enableLocalization, "true", StringComparison.OrdinalIgnoreCase) == 0 ?

Reply:
better to test out with Off Flag as well to make sure it works for S-Clouds :)

Reply:
I thought we are gonna enable this by default. But Shekhar you are right, the default behavior should be OFF instead. I will test this:)